rclcpp_lifecycle/include/rclcpp_lifecycle/lifecycle_node.hpp

```diff
@@ -268,10 +287,58 @@ class LifecycleNode : public node_interfaces::LifecycleNodeInterface,
     const std::string & name,
     rclcpp::Parameter & parameter) const;
 
+  RCLCPP_LIFECYCLE_PUBLIC
```

Comment:
you don't need that for template functions because templates are getting resolved on compile time in the caller library and don't have linkage visibility.

Reply:
same for the template functions below.

Reply:
@vinnamkim Can you please update to fix the issues that Karsten mentions?

Reply:
@Karsten1987, @mjeronimo  Removed macros and updated commit.